deluge/ui/gtk3/piecesbar.py

```diff
@@ -152,22 +155,21 @@ def write_text(self):
 
         if self.resized() or self.text != self.prev_text or self.text_overlay is None:
             # Need to recreate the cache drawing
-            self.text_overlay = ImageSurface(FORMAT_ARGB32, self.width, self.height)
-            ctx = Context(self.text_overlay)
-            pg = CairoContext(ctx)
-            pl = pg.create_layout()
+            self.text_overlay = cairo.ImageSurface(cairo.FORMAT_ARGB32, self.width, self.height)
+            ctx = cairo.Context(self.text_overlay)
+            pl = PangoCairo.create_layout(ctx)
             pl.set_font_description(self.text_font)
             pl.set_width(-1)  # No text wrapping
-            pl.set_text(self.text)
+            pl.set_text(self.text, len(self.text))
```

Comment:
I think `-1` for length parameter would suffice here: https://developer.gnome.org/pango/stable/pango-Layout-Objects.html#pango-layout-set-text